### DISCUSSION TITLE
No output from python UDF

### DISCUSSION THREAD

**Discussion by belford33** [davidfrantz/force#318] [Q&A] [answered]
Hello David, I am facing a similar problem as the previous writer (thielfab). I would like to test the UDF mowing_detection.py and have tried several FORCE versions (v3710, 3712 and dev) but the output is always empty. Any idea how to solve this? Let me know if you need more infos from me. Thanks a lot and all the best, Thomas

**Comment by davidfrantz** [comment]
Dear Thomas,

I can't reproduce this here. Can you please share your parameter file?

And please also clarify whether an image with only nodata values is written, or if nothing is written at all?

Cheers,
David

**Reply by belford33** [reply]
Thanks David for your reply and I am happy to provide more information. I can confirm that the mowing detection code is running and the output is written to the drive but the entire area has the nodata value. 
Please find here the parameter file which I used in the developer version. 
[mowing_dev.txt](https://github.com/davidfrantz/force/files/14863703/mowing_dev.txt)
Let me know if that helps,
thanks a lot,
best,
Thomas

**Comment by davidfrantz** [comment]
Most likely, some of the filters result in no data being left for analysis.

What does the tool say?
Was input detected?

**Comment by ernstste** [comment]
My message earlier apparently didn't go through, here it is again:

```
Hi Thomas,

the mowing detection code has some imperfections with date conversions. Check if you have images for the first few days of the year. If so, remove them (they don't bring value to the analysis if you're on the Northern Hemisphere anyways).
Another common source of error is accidentally using a mask that excludes your areas of interest.

Stefan
```

**Reply by belford33** [reply]
Dear David, dear Stefan,
 
I excluded January and February images by setting the DOY range accordingly in the parameter file and I get the desired results. :) I confirm that this must be related to the dates which are outside the range used by the mowing detection algorithm. Thanks a lot for your quick and good support! All the best, Thomas

**Reply by ernstste** [reply, marked as answer]
Great!

Simply removing the first scene of the year from your datacube (probably on the first of January?) should be enough. If I remember correctly there was a rounding issue that can lead to the mowing detection code only considering scenes from the prior year if the first scene is very close to the beginning of the year.
